File: c++/048.cpp
class Solution {
/*
我的思路：
1. 以矩阵对角线为对称轴，交换对称位置的值
2. 以矩阵竖对称轴为轴，交换对称位置的值
*/
public:
    void rotate(vector<vector<int>>& matrix) {
        int n=matrix.size();
        for(int i=0; i<n; i++)
            for(int j=i+1; j<n; j++)
                swap(matrix[i][j],matrix[j][i]);
        for(int i=0; i<n; i++)
            for(int j=0,k=n-1;j<k;j++,k--)
                swap(matrix[i][j],matrix[i][k]);
    }
};

File: go/048.go
package main

/**
旋转图像
思路: 对角线翻转 + 竖轴翻转
*/
func rotate(matrix [][]int) {
	l := len(matrix)
	for i := 0; i < l; i++ {
		for j := i + 1; j < l; j++ {
			matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
		}
	}
	for i := 0; i < l; i++ {
		for j := 0; j < l/2; j++ {
			matrix[i][j], matrix[i][l-j-1] = matrix[i][l-j-1], matrix[i][j]
		}
	}
}
